src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -207,7 +207,24 @@ public override Task FlushAsync(CancellationToken cancellationToken)
             Interlocked.Increment(ref _asyncOperations);
             try
             {
-                await base.FlushAsync(cancellationToken).ConfigureAwait(false);
+                EnsureNotDisposed();
```

Comment:
We just checked EnsureNotDisposed in the caller... I don't think we need to do so again here.